h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -248,14 +379,7 @@ private void gridSearch(Grid<MP> grid) {
           grid.update(_job);
           final MP generalParams = _hyperSpaceWalker.getParams();
           if (generalParams._export_checkpoints_dir != null) {
```

Comment:
I'd remove this condition for clarity as it is repeated in the `attemptGridSave` method.